From 40257d65b338b6e2ed9d89d6fa7c7b8701a4c311 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Wed, 6 Aug 2014 14:54:12 -0400
Subject: [PATCH 1/2] defn2[c|man]: don't rely on dpkg-architecture to set arch

In yocto we'll always be cross compiling, and we'll always
be building on linux for linux (vs. *BSD, hurd, etc.)

Without this the arch is not detected, but it doesn't error
out, and hence you get useless binaries that don't know any
arch specific methods, and the end result will be strangeness
like the loopback device not being configured/enabled.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Upstream-Status: Pending
---
 defn2c.pl   | 6 +++---
 defn2man.pl | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/defn2c.pl b/defn2c.pl
index 8798dc2..f2551c7 100755
--- a/defn2c.pl
+++ b/defn2c.pl
@@ -2,9 +2,9 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
-
-$DEB_HOST_ARCH_OS =~ s/\n//;
+#my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+#$DEB_HOST_ARCH_OS =~ s/\n//;
+my $DEB_HOST_ARCH_OS ="linux";
 
 # declarations
 my $address_family = "";
diff --git a/defn2man.pl b/defn2man.pl
index 6ddcfdd..c9c4dd0 100755
--- a/defn2man.pl
+++ b/defn2man.pl
@@ -2,9 +2,9 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
-
-$DEB_HOST_ARCH_OS =~ s/\n//;
+#my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+#$DEB_HOST_ARCH_OS =~ s/\n//;
+my $DEB_HOST_ARCH_OS = "linux";
 
 # declarations
 my $line;
-- 
2.17.1

